brave/src/main/java/com/linecorp/armeria/common/brave/RequestContextCurrentTraceContext.java

```diff
@@ -101,7 +106,41 @@ public String toString() {
         }
     };
 
-    static final class Builder extends CurrentTraceContext.Builder {
+    public static final class Builder extends CurrentTraceContext.Builder {
+
+        private final ImmutableList.Builder<Pattern> nonRequestThreadPatterns = ImmutableList.builder();
+
+        /**
+         * Sets a regular expression that matches names of threads that should be considered non-request
+         * threads, meaning they may have spans created for clients outside of the context of an Armeria
+         * request. For example, this can be set to {@code "^RMI TCP Connection.*"} if you use RMI to serve
+         * monitoring requests.
+         *
+         * @see #setCurrentThreadNotRequestThread(boolean)
+         */
+        public Builder nonRequestThread(String pattern) {
+            requireNonNull(pattern, "pattern");
+            final Pattern compiled;
+            try {
+                compiled = Pattern.compile(pattern);
+            } catch (PatternSyntaxException e) {
+                throw new IllegalArgumentException("Invalid regex: " + pattern);
+            }
```

Comment:
`PatternSyntaxException` is already an `IllegalArgumentException`, so maybe we don't need to catch?